x-pack/plugins/triggers_actions_ui/public/application/sections/common/components/with_bulk_alert_api_operations.tsx

```diff
@@ -31,14 +30,24 @@ export interface ComponentOpts {
   unmuteAlerts: (alerts: Alert[]) => Promise<void>;
   enableAlerts: (alerts: Alert[]) => Promise<void>;
   disableAlerts: (alerts: Alert[]) => Promise<void>;
-  deleteAlerts: (alerts: Alert[]) => Promise<void>;
+  deleteAlerts: (
+    alerts: Alert[]
+  ) => Promise<{
+    successes: string[];
+    errors: string[];
+  }>;
   muteAlert: (alert: Alert) => Promise<void>;
   unmuteAlert: (alert: Alert) => Promise<void>;
   muteAlertInstance: (alert: Alert, alertInstanceId: string) => Promise<void>;
   unmuteAlertInstance: (alert: Alert, alertInstanceId: string) => Promise<void>;
   enableAlert: (alert: Alert) => Promise<void>;
   disableAlert: (alert: Alert) => Promise<void>;
-  deleteAlert: (alert: Alert) => Promise<void>;
+  deleteAlert: (
+    alert: Alert
+  ) => Promise<{
+    successes: string[];
+    errors: string[];
+  }>;
```

Comment:
I like that we're using only one api for deleting (using `deleteAlerts` for both multiple and single deletion requests), but I have a little _nit picky_ thought. :)
It feels weird returning `{ successes: string[]; errors: string[]; }` when the user calls `deleteAlert`, as it's only one id.

Could we perhaps change the signature to `{ successes?: string; errors?: string;}`?

Reply:
If we're going to inline the type as is done here, agree w/Gidi that we could change the structure to match the semantic value.  On the other hand, I can see this pattern of successes/errors being more widely used, and probably worth a separate type itself, and then having two different types - a "single" and "multiple" version - could get unwieldy.